troubleshoot-data-inconsistency-errors.md

```diff
@@ -0,0 +1,93 @@
+
+#### Error 8141
+
+`ERROR 8141 (HY000): assertion failed: key: 7480000000000000405f72013300000000000000f8, assertion: NotExist, start_ts: 430590532931813377, existing start ts: 430590532931551233, existing commit ts: 430590532931551234`
+
+This error indicates that the assertion failed when a transaction was being committed. Assuming that data and index are consistent, TiDB asserted that the key `7480000000000000405f720133000000000000000000f8` did not exist. When the transaction was being committed, TiDB found the key did exist, written by the transaction with the `start ts` of `430590532931551233`. TiDB will print the MVCC (Multi-Version Concurrency Control) history of this key to logs.
+
+### Errors reported in admin check
+
+This section lists the data inconsistency errors that might occur in TiDB when you execute the [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) statement, and explains the meanings of these errors with examples.
+
+#### Error 8003
+
+`ERROR 8003 (HY000): table count 3 != index(idx) count 2`
+
+This error indicates that the table on which the `ADMIN CHECK` statement is executed has 3 row key-value pairs but only 2 index key-value pairs.
+
+#### Error 8134
+
+`ERROR 8134 (HY000): data inconsistency in table: t, index: c2, col: c2, handle: "2", index-values:"KindInt64 13" != record-values:"KindInt64 12", compare err:<nil>`
+
+This error indicates that for the `c2` index in table `t`, the handle of a row is 13 in the index key-value pair but is 12 in the row record key-value pair, which results in data inconsistency.
+
+#### Error 8223
+
+`ERROR 8223 (HY000): data inconsistency in table: t2, index: i1, handle: {hello, hello}, index-values:"" != record-values:"handle: {hello, hello}, values: [KindString hello KindString hello]"`
+
+This error indicates that `index-value` is null and `record-values` is not null, meaning that the record does not have the corresponding index but has the corresponding row, which results in data inconsistency.
+
+## Reasons and solutions
+
+When a data consistency error occurs, the reasons can be as follows:
+
+- The data and index in the existing data are consistent and the current version of TiDB has a bug. If an ongoing transaction is about to write inconsistent data, TiDB aborts the transaction.
+- The data and index in the existing data are inconsistent. The inconsistent data could be from an error or a dangerous operation in the past or caused by a TiDB bug.
+- The data and index are consistent but the detection algorithm has a bug that causes errors by mistake.
+
+If you receive a data inconsistency error, contact PingCAP technical support immediately to troubleshoot or fix it instead of dealing with the error by yourself. If PingCAP technical support confirms that the error is reported by mistake, or your application needs to skip such errors urgently, you can use the following methods to bypass the check.
```

Comment:
```suggestion
If you receive a data inconsistency error, contact PingCAP technical support for troubleshooting immediately instead of dealing with the error by yourself. If PingCAP technical support confirms that the error is reported by mistake, or your application needs to skip such errors urgently, you can use the following methods to bypass the check.
```